### PR TITLE
Started a re-write of the docs

### PR DESCRIPTION
**Pull request by Torxed**
Using CSV for tables and re-organizing parameter definitions
Organizing the layout a bit better.

And I'll add more concrete examples of how to run different scenarios *(such as re-using `/boot`, separating `/efi` and dual-booting scenarios)*

This fixes:
 * #2223
 * #2144 *(partially)*
 * #2002 *(partially)*

### CODE REVIEW

File: docs/installing/config_options.csv [hunk 1]
```diff
@@ -0,0 +1,24 @@
+Key,Value(s),Description,Required
+additional-repositories,[ `multilib <https://wiki.archlinux.org/title/Official_repositories#multilib>`_!, `testing <https://wiki.archlinux.org/title/Official_repositories#Testing_repositories>`_ ],Enables one or more of the testing and multilib repositories before proceeding with installation,No
+archinstall-language,`lang <https://github.com/archlinux/archinstall/blob/master/archinstall/locales/languages.json>`_,Sets the TUI language used *(make sure to use the ``lang`` value not the ``abbr``)*,No
+audio_config,`pipewire <https://wiki.archlinux.org/title/PipeWire>`_!, `pulseaudio <https://wiki.archlinux.org/title/PulseAudio>`_,Audioserver to be installed,No
+bootloader,`Systemd-boot <https://wiki.archlinux.org/title/Systemd-boot>`_!, `grub <https://wiki.archlinux.org/title/GRUB>`_,Bootloader to be installed *(grub being mandatory on BIOS machines)*,Yes
+debug,``true``!, ``false``,Enables debug output,No
+disk_config,*`read more under Disk Configuration <#>`_*,Contains the desired disk setup to be used during installation,No
```
Comment (alan-barzilay) [inline docs/installing/config_options.csv:7]: hi @Torxed, is there more information about disk_config anywhere? 
I'm trying to reverse engineer the parameters from running it guided but I haven't had much success haha I stumbled on this branch/PR but couldn't find anything else about disk_config here, that's a section that will be created/updated in the future? Thanks for your time and the amazing tool!

Comment (Torxed) [reply docs/installing/config_options.csv:7]: I can make sure to update `disk_config` before anything else :) I'm expecting to be working on this during the weekend quite a bit, but can *try* (no promises) to get `disk_config` done before that a :)

Comment (alan-barzilay) [reply docs/installing/config_options.csv:7]: I would really appreciate it haha Anything I could do to help move things forward? I can try to help to write parts of the documentation I understand but Im not sure if my help would increase or decrease your productivity haha

Comment (Torxed) [reply docs/installing/config_options.csv]: @alan-barzilay I've submitted a decent first draft of how disk partitioning works and the different modes. You'll find them under `archinstall/docs/_build/html/cli_parameters/config/disk_config.html` after running:

```shell
$ cd archinstall/docs
$ make clean html
```

They'll get pushed to the readthedocs once I'm finished here :)

File: docs/cli_parameters/config/disk_config.rst [hunk 2]
```diff
@@ -0,0 +1,244 @@
+.. _disk config:
+
+Disk Configuration
+==================
+
+There are only three modes in the ``disk_config`` option. They are described in more detail below.
```
Comment (alan-barzilay) [inline docs/cli_parameters/config/disk_config.rst:6]: @Torxed shouldnt it be 4 modes? "leave as is" + the three from the guided menu (Best Effort, 
Manual Partitioning and *pre-mounted configuration*).

Or leave as is and pre-mounted configuration are the same in some way?

Comment (Torxed) [reply docs/cli_parameters/config/disk_config.rst:6]: > Or leave as is and pre-mounted configuration are the same in some way?

Correct :) They're identical, It's just me using different names for the same thing. I should unify those names! Thanks for pointing it out!

**Comment by alan-barzilay** [comment]
update with changes brought by #2221

**Comment by Torxed** [comment]
> update with changes brought by #2221

Done.

**Comment by alan-barzilay** [comment]
it would be nice to take this opportunity to spruce up the readme, something like #2174 suggests

File: docs/installing/guided.rst [hunk 3]
```diff
@@ -240,20 +247,41 @@ Options for ``--creds``
...
+
+
+.. note::
+
+   ``!users`` is optional only if ``!root-password`` was set. ``!users`` will be enforced otherwise and the minimum amount of users with sudo privileges required will be set to 1.
 
 .. note::
-    ``!users`` is optional only if ``!root-password`` was set. ``!users`` will be enforced otherwise and the minimum amount of users with sudo privileges required will be set to 1.
 
+   The key's start with ``!`` because internal log functions will mask any keys starting with explamation from logs and unrestricted configurations.
```
[the remark this answers is not in the document]
Comment (Torxed) [reply docs/installing/guided.rst:284]: There's going to be ~50+ ^^

**Comment by Torxed** [comment]
> it would be nice to take this opportunity to spruce up the readme, something like #2174 suggests

I still have to think about that one, especially the phrasing that's changed quite a bit.
I'm not against it by any means, but I'd like to carefully consider the changes. But yes, any updates is only going to improve things heh.

**Pull request merged**